### PR TITLE
Add hex numbers with uppercase X to grammar.txt

### PR DESCRIPTION
**Pull request by federicobond**

### CODE REVIEW

File: docs/grammar.txt [hunk 1]
```diff
@@ -131,7 +131,7 @@ HexLiteral = 'hex' ('"' ([0-9a-fA-F]{2})* '"' | '\'' ([0-9a-fA-F]{2})* '\'')
 StringLiteral = '"' ([^"\r\n\\] | '\\' .)* '"'
 Identifier = [a-zA-Z_$] [a-zA-Z_$0-9]*
 
-HexNumber = '0x' [0-9a-fA-F]+
+HexNumber = '0' [xX] [0-9a-fA-F]+
```
Comment (ekpyron) [inline docs/grammar.txt:134]: While we're at it, we should probably adjust it further to reflect the fact that we now allow underscores in number literals (see #4684).

Comment (chriseth) [reply docs/grammar.txt:134]: I'll still merge this for now.

Comment (axic) [reply docs/grammar.txt:134]: What is the reason to accept uppercase X btw? It is kind of invalid in a lot of languages.

Comment (chriseth) [reply docs/grammar.txt:134]: I'm fine with disallowing it. I think it is a remnant of the v8 scanner.

Comment (axic) [reply docs/grammar.txt:134]: Created an issue: https://github.com/ethereum/solidity/issues/4899

**Pull request merged**